### PR TITLE
Issue #10164: Refactor attribute to property

### LINKED ISSUE #10164: Add Configuration.getPropertyNames and Configuration.getProperty and deprecate getAttributeNames getAttribute methods
there is no Atributes in Checkstyle, we have properties.
https://checkstyle.org/config.html#Properties

Attribute term probably come to us from XML config, as the only way to configure checkstyle.
But we consider other way of doing it configs: yml, .... .

API should be consistent with general documentation, and implementation details should not be leaked in API.

detected at https://github.com/checkstyle/checkstyle/pull/10142#discussion_r657443405

### PR DESCRIPTION
**Pull request by shashwatj07**
Resolves #10164

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/api/Configuration.java [hunk 1]
```diff
@@ -30,20 +30,20 @@
...
 
     /**
-     * The set of attribute names.
+     * The set of property names.
      *
-     * @return The set of attribute names, never null.
+     * @return The set of property names, never null.
      */
-    String[] getAttributeNames();
+    String[] getPropertyNames();
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/Configuration.java:37]: this is API, we can not damage it, CI failures are probably is proving this.

Please create new methods `xxxPropertyXxxx` and deprecate  `xxxAttributeXxxx` (xxxAttributeNames will simply call xxxPropertyXxxx in implementation).
Whole checkstyle should migrate to `xxxPropertyXxxx`. Later in future we will remove xxxAttributexxxx.

**Review by romani — CHANGES_REQUESTED** [review]
Ci should pass
item:

File: src/main/java/com/puppycrawl/tools/checkstyle/api/Configuration.java (hunk 1 above)
Comment (shashwatj07) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/Configuration.java:59]: done, please take another look

**Review by romani — APPROVED** [review]
approved with minor item to fix:

File: .ci/jsoref-spellchecker/whitelist.words [hunk 2]
```diff
@@ -117,6 +117,7 @@ beanutils
 beforeexecutionexclusionfilefilter
 bestpractices
 bew
+BFC
```
[the remark this answers is not in the document]
Comment (shashwatj07) [reply .ci/jsoref-spellchecker/whitelist.words]: done

**Review by pbludov — APPROVED** [review]

**Pull request merged**